src/Definition/Reflection/ClassDefinition.php

```diff
@@ -34,7 +33,10 @@ public function __construct($class)
         $this->reflection = $class;
     }
 
-    private function reflectSupertypes()
+    /**
+     * @psalm-assert list<string> $this->supertypes
```

Comment:
instead of this, wouldn't it be better to simply set `supertypes` to array with a default value?

Reply:
Probably yes, but this would require additional refactoring to this class since `null` is used by getter methods. I just added the types to reflect the existing behaviour.